Testing/functional/tests/cypress/integration/ui/user/appTour.js

```diff
@@ -0,0 +1,87 @@
+        cy.intercept("GET", `**/UserProfile/${hdid}`, {
+            fixture: "UserProfileService/userProfile.json",
+        });
+        cy.login(
+            Cypress.env("keycloak.username"),
+            Cypress.env("keycloak.password"),
+            AuthMethod.KeyCloak,
+            "/home"
+        );
+    });
+
+    it("Header tour button should be visible and have badge when tour changes present", () => {
+        cy.get("[data-testid=app-tour-button]")
+            .should("be.visible")
+            .find("span")
+            .should("have.class", "b-avatar-badge badge-danger");
+
+        cy.get("[data-testid=app-tour-button").click();
+
+        cy.get("[data-testid=app-tour-modal]").should("be.visible");
+
+        // Badge should be removed once tour is opened
+        cy.get("[data-testid=app-tour-button")
+            .find("span.b-avatar-badge")
+            .should("not.exist");
+    });
+
+    function keepClickNext() {
+        const nextButtonId = "[data-testid=app-tour-next]";
+        cy.get("[data-testid=app-tour-modal]").then((modal) => {
+            if (modal.find(nextButtonId).length) {
+                cy.get(nextButtonId).should("be.visible").click();
+                keepClickNext();
+            }
+        });
+    }
+
+    it("Modal controls should be visible and work (no back)", () => {
+        cy.get("[data-testid=app-tour-button]").click();
+        cy.get("[data-testid=app-tour-modal]").should("be.visible");
+
+        cy.get("[data-testid=app-tour-back]")
+            .should("be.visible")
+            .should("be.disabled");
+
+        cy.get("[data-testid=app-tour-skip]").should("be.visible").click();
+        cy.get("[data-testid=app-tour-modal]").should("not.exist");
+
+        cy.get("[data-testid=app-tour-button]").click();
+
+        keepClickNext();
+
+        // last slide app-tour-done should be visible
+        cy.get("[data-testid=app-tour-done]").should("be.visible").click();
+
+        // modal should be closed
+        cy.get("[data-testid=app-tour-modal]").should("not.exist");
+    });
+
+    // This greatly depends on the tour content, and isn't a great test but will test if the content is there
+    it("Modal controls should be visible and work forwards and backwards", () => {
+        cy.get("[data-testid=app-tour-button]").click();
+        cy.get("[data-testid=app-tour-modal]").should("be.visible");
+
+        // first slide back should be disabled
+        cy.get("[data-testid=app-tour-back]")
+            .should("be.visible")
+            .should("be.disabled");
+
+        cy.get("[data-testid=app-tour-next]").should("be.visible").click();
+
+        cy.get("[data-testid=app-tour-modal]").then((modal) => {
+            if (modal.find("[data-testid=app-tour-back]").length) {
+                cy.get("[data-testid=app-tour-back]")
+                    .should("be.visible")
+                    .click();
+            }
+            cy.log("No back button found");
+        });
```

Comment:
This part of the test isn't doing much.
```suggestion
        cy.get("[data-testid=app-tour-back]")
            .should("be.visible")
            .should("be.enabled")
            .click();

        cy.get("[data-testid=app-tour-back]")
            .should("be.visible")
            .should("be.disabled");
```

Reply:
This is based on the configuration of the slides, if there is per se only 2 slides then there'll never be an active back button.

its conditional to manage the changing nature of the slides, if it exists it checks the availability and the click, if it doesn't a better check would be to ensure a done button is available.

I'll update the test to better reflect this but your suggestion would break the test if there is only 2 slides, not likely but possible.

Reply:
I don't think we need to worry about there being only 2 slides. If there were only 2 slides, we wouldn't need a back button at all and this test could be removed.

Reply:
I've amended to test the requirements more robustly.